sphinx/ext/collapse.py

```diff
@@ -0,0 +1,206 @@
+"""Support for collapsible content in HTML."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, ClassVar
+
+from docutils import nodes
+from docutils.parsers.rst import directives
+
+import sphinx
+from sphinx.transforms.post_transforms import SphinxPostTransform
+from sphinx.util.docutils import SphinxDirective
+
+if TYPE_CHECKING:
+    from docutils.parsers.rst.states import RSTState
+    from docutils.statemachine import StringList
+
+    from sphinx.application import Sphinx
+    from sphinx.util.typing import ExtensionMetadata, OptionSpec
+    from sphinx.writers.html5 import HTML5Translator
+
+
+class collapsible(nodes.General, nodes.Element):
+    """Node for collapsible content.
+
+    This is used by the :rst:dir:`collapse` directive.
+    """
+
+
+class summary(nodes.General, nodes.TextElement):
+    """Node for the description for collapsible content.
+
+    This is used by the :rst:dir:`collapse` directive.
+    """
+
+
+def visit_collapsible(translator: HTML5Translator, node: nodes.Element) -> None:
+    if node.get('collapsible_open'):
+        translator.body.append(translator.starttag(node, 'details', open='open'))
```

Comment:
It would be good to add a class to this, something like `sphinx-ext-collapse`, so that they can be targeted more specifically in css.